kv/task.go

```diff
@@ -267,7 +267,7 @@ func (s *Service) findTasksByUser(ctx context.Context, tx Tx, filter influxdb.Ta
 			continue
 		}
 
-		if taskFilterMatch(filter.Type, task.Type) {
+		if taskTypeFilterMatch(filter.Type, task.Type) && taskStatusFilterMatch(filter.Active, task.Status) {
```

Comment:
We can pair and explore whether there is an opportunity to simplify the `findTask` APIs to use the `newTaskMatchFn` as you have expanded this already.